python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -431,10 +425,15 @@ def categorical_indices_from_vocabulary_file(feature_info, feature_tensor, file_
 
     if "dropout_rate" in args:
         default_value = 0  # Default OOV index when using dropout
+        if args.get("num_oov_buckets"):
+            raise RuntimeError("Cannot have both dropout_rate and num_oov_buckets set. "
+                               "OOV buckets are not supported with dropout")
```

Comment:
Enforce that we don't have both at the same time.

Reply:
Do we know why? I had the impression that there was an issue with this Ashish was working on, but is it confirmed?

Reply:
I don't know. This just reflect what is already in the code.

Reply:
By the nature of how "dropout" is defined on categorical labels, you can only assign it to only 1 bucket. The reason I introduced this notion of a dropout on categorical labels is to make sure that each category contributes to the OOV embedding(instead of only low count categories). (I'm trying to be really generic in my wording here haha)

But yeah, we can modify it to use multiple buckets. Feel free to do it if you want.
Check out how the VocabLoopkup differs for the two cases.

Reply:
Essentially, I needed a neat way to "fix" an OOV token. I can't do it easily with `StaticVocabularyTable` as it depends on number of unique keys and such. Again, can be done in some way. But wasn't really critical.